tests/tcp.rs

```diff
@@ -23,7 +23,99 @@ fn assert_error_kind<T>(res: io::Result<T>, kind: io::ErrorKind) {
 }
 
 async fn bind() -> std::result::Result<TcpListener, std::io::Error> {
-    TcpListener::bind((IpAddr::from(Ipv4Addr::UNSPECIFIED), PORT)).await
+    TcpListener::bind((Ipv4Addr::UNSPECIFIED, PORT)).await
+}
+
+#[test]
+fn connects_within_a_host() -> Result {
+    let mut sim = Builder::new().build();
+
+    sim.client("server", async {
+        let listener = bind().await?;
+        tokio::spawn(async move {
+            loop {
+                let _ = listener.accept().await;
+            }
+        });
+
+        TcpStream::connect(("server", PORT)).await?;
```

Comment:
I see, so you did (addr, addr) to make this work. Do we need it?

Reply:
Actually, I don't think we don't. I'll remove it. If we need it in the future we can always add it.